text/0020-dispersion-payload.md

```diff
@@ -0,0 +1,77 @@
+
+### Test Cases
+
+A procotol can be very difficult to analyze in isolation so I started by creating the following set of test cases. These test cases are effectively situations and workflows the protocol must support to provide any value. I then used the proposed protocol to accomplish said test cases. This is an excellent review point for industry specific individuals who might not have a technical background. For those with a technical background this is still a useful starting point because it provides context.
+
+[x] user needs to manually rinse out the system and must be able to control via some sort of interface (physical or digital)
+[x] user might need to stop spray mid flight for an emergency without the autopilot being able to override
+[x] payload gets overfilled during servicing
+[x] payload is not full enough when trying to take off
+[x] clogged nozzle during flight
+[x] motor (spray pump, or spreader motor) failure during flight
+[x] spray system leaks
+[x] autopilot control of spray system
+[x] aircraft exceeds speed at which the device can keep up
+[x] device is turned on before aircraft is at minimum speed
+[x] dynamically adjust deposition rate based on aircraft speed
+[x] get payload information for planning (ie lead in/out distances, spray leg length, etc)
+
+ - Swath width will be configured during planning so OOS
+
+[x] data for compliance and efficancy reporting can be extracted
+
+ - Factors that cannot be known by the dispersion device is OOS and can be collected in other parts of the system (ie wind, chemical being applied, etc)
+
+[x] payload is at target fill and ready to go
+[x] spot spraying. This happens at the planning level. This protocol supports setting flow rate at specific locations which is enough for spot spraying.
+[x] communication buses might be lossy so protocol must handle failure detection.
+
+### Implementation
+
+After reviewing other submissions, I think I might have gone too far down this path but I wanted something complete to get reviewed by industry relevant people in my network before submitting here to the MAVLink community. I created a messages.md file with the full set of message definitions and a microservice.md file detailing how the protocol should work that is submitted with this PR. I have a couple of versions: one with a manager and one without. I used the one without because it is less complex and conveys the same concept.
+
+### Questions
+
+1. This is a combined protocol for spraying and spreading since there are many common elements. Should this be split up?
```

Comment:
I think that it makes sense to combine it as you have. The downside of this from a MAVLink point of view is that it is annoying to have two units for one field. We might add enums to indicate units in use.